lib/gst/player/gstplayer-media-info.c

```diff
@@ -0,0 +1,562 @@
+#include "gstplayer-media-info.h"
+#include "gstplayer-media-info-private.h"
+
+/* Per-stream information */
+G_DEFINE_TYPE (GstPlayerStreamInfo, gst_player_stream_info, G_TYPE_OBJECT);
+
+static void
+gst_player_stream_info_init (GstPlayerStreamInfo * sinfo)
+{
+  /* Nothing to do here */
+}
+
+static void
+gst_player_stream_info_finalize (GObject * object)
+{
+  GstPlayerStreamInfo *sinfo = GST_PLAYER_STREAM_INFO (object);
+
+  if (sinfo->caps) {
+    gst_caps_unref (sinfo->caps);
+    sinfo->caps = NULL;
+  }
+
+  if (sinfo->tags) {
+    gst_tag_list_unref (sinfo->tags);
+    sinfo->tags = NULL;
+  }
+}
+
+static void
+gst_player_stream_info_class_init (GObjectClass * klass)
+{
+  klass->finalize = gst_player_stream_info_finalize;
+}
+
+/**
+ * gst_player_stream_info_get_stream_id:
+ * @info: a #GstPlayerStreamInfo
+ *
+ * Returns: the stream ID of this stream.
+ */
+gint
+gst_player_stream_info_get_stream_id (const GstPlayerStreamInfo *info)
+{
+  return info->stream_id;
+}
+
+/**
+ * gst_player_stream_info_get_stream_type_nick:
+ * @info: a #GstPlayerStreamInfo
+ *
+ * Returns: a human readable name for the stream type of the given @info (ex : "audio",
+ * "video",...).
+ */
+const gchar*
+gst_player_stream_info_get_stream_type_nick (const GstPlayerStreamInfo *info)
+{
+  if (GST_IS_PLAYER_VIDEO_INFO (info))
+    return "video";
+
+  if (GST_IS_PLAYER_AUDIO_INFO (info))
+    return "audio";
+
+  if (GST_IS_PLAYER_SUBTITLE_INFO (info))
+    return "subtitle";
+
+  return NULL;
+}
+
+/**
+ * gst_player_stream_info_get_tags:
+ * @info: a #GstPlayerStreamInfo
+ *
+ * Returns: the tags contained in this stream. Unref with #gst_tag_list_unref
+ */
+GstTagList*
+gst_player_stream_info_get_stream_tags (const GstPlayerStreamInfo *sinfo)
+{
+  GstTagList *tags = NULL;
+
+  if (sinfo->tags)
+    tags = gst_tag_list_ref (sinfo->tags);
+
+  return tags;
+}
+
+/**
+ * gst_player_stream_info_get_caps:
+ * @info: a #GstPlayerStreamInfo
+ *
+ * Returns: the #GstCaps of the stream. Unref with #gst_caps_unref after usage.
+ */
+GstCaps*
+gst_player_stream_info_get_stream_caps (const GstPlayerStreamInfo *sinfo)
+{
+  GstCaps *caps = NULL;
+
+  if (sinfo->caps)
+    caps = gst_caps_ref (sinfo->caps);
+
+  return caps;
+}
+
+/**
+ * gst_player_video_info_get_streams:
+ * @info: a #GstPlayerMediaInfo
+ *
+ * Returns: (transfer full) (element-type GstPlayerVideoInfo): A #GList of
+ * matching #GstPlayerVideoInfo. The caller should free it with
+ * gst_player_stream_info_list_free().
+ */
+GList*
+gst_player_media_info_get_video_streams (const GstPlayerMediaInfo *info)
+{
+  GList *result = NULL, *p;
+
+  g_return_val_if_fail (GST_IS_PLAYER_MEDIA_INFO (info), NULL);
+
+  for (p = info->stream_list; p != NULL; p = p->next) {
+    GstPlayerStreamInfo  *sinfo = (GstPlayerStreamInfo*) p->data;
+
+    if (GST_IS_PLAYER_VIDEO_INFO (sinfo)) {
+      GstPlayerVideoInfo  *vinfo = (GstPlayerVideoInfo*) sinfo;
+
+      result = g_list_append (result, vinfo);
```

Comment:
As you say this is transfer-full (which is the right thing to do), you need to store a new reference in the returned list.